client/src/shared/components/Dropdown/index.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from 'react';
+
+import ChevronDown from '@/src/shared/icons/ChevronDown';
+
+interface DropdownProps {
+  options: string[]
+  label: string
+}
+
+const Dropdown: React.FC<DropdownProps> = ({
+  options,
+  label
+}: DropdownProps) => {
+  const [selectedOption, setSelectedOption] = useState<string | undefined>(
+    undefined
+  );
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleOptionSelectEvent = (option: string): void => {
+    setSelectedOption(option);
+    setIsOpen(false);
+  };
+
+  return (
+    <div className="relative inline-block text-left">
+      <div>
+        <button
+          type="button"
+          className="inline-flex justify-between w-full rounded-md border border-gray-300 shadow-sm px-4 py-2 bg-white text-sm font-medium text-gray-700 hover:bg-gray-50 focus:outline-none"
+          id="options-menu"
+          aria-haspopup="true"
+          aria-expanded="true"
+          onClick={() => {
+            setIsOpen(!isOpen);
+          }}
+        >
+          <span>{selectedOption ?? label}</span>
+          <div className="-mr-1 ml-2 h-5 w-5" aria-hidden="true">
+            <ChevronDown />
+          </div>
+        </button>
+      </div>
+
+      {isOpen && (
+        <div
+          className="origin-top-right absolute mt-2 w-56 rounded-md shadow-lg bg-white ring-black ring-opacity-5 focus:outline-none"
+          role="menu"
+          aria-orientation="vertical"
+          aria-labelledby="options-menu"
+        >
+          <div className="py-1" role="none">
+            {options.map((option) => (
+              <a
+                key={option}
+                href={`#${option}`}
+                className={`${
+                  selectedOption === option
+                    ? 'bg-gray-100 text-gray-900'
+                    : 'text-gray-700'
+                } block px-4 py-2 text-sm w-full text-left hover:bg-gray-100`}
+                role="menuitem"
+                onClick={() => {
+                  handleOptionSelectEvent(option);
+                }}
+              >
+                {option}
+              </a>
+            ))}
+          </div>
+        </div>
+      )}
+    </div>
+  );
+};
+
+Dropdown.defaultProps = {
+  options: ['option1', 'option2', 'option3'],
+  label: 'Dropdown menu'
```

Comment:
- while `label` should be optional. and should have just an empty string as default value